### ISSUE TITLE
Selfhosted : Transfer ownership results in "500" error

### ISSUE THREAD

**Issue by tmrwltd** [plausible/analytics#1545]
### Past Issues Searched

- [X] I have searched open and closed issues to make sure that the bug has not yet been reported

### Issue is a Bug Report

- [X] This is a bug report and not a feature request, nor asking for self-hosted support

### Describe the bug

Try to transfer an ownership results in an "Error 500". But the transfer is initiated. The new user can accept - but also results in an "Error 500". But also in this case, the ownership is changed successfully.
But the new owner always get an message, that he needs an subscription to access the sites he own.

### Expected behavior

No "500" errors and no "Payment" message coz it's an selfhosted Plausible instance.

### Screenshots

![grafik](https://user-images.githubusercontent.com/50417351/146496894-d3909eb5-7af2-4ce8-8377-9c92f22cd0ea.png)

![grafik](https://user-images.githubusercontent.com/50417351/146497187-ff957fb8-1047-40b3-848c-5fcf7b5ce625.png)

### Environment

```markdown
- OS: Windows 11

- Browser: Firefox Developer
- Browser Version: 96.0b6

- Browser: Firefox
- Browser Version: 95.0.1
```

**Comment by ukutaht** [comment]
Hey, thanks for reporting.

1. The ownership transfer request results in 500 probably because it attempts to send an email but is unable to do so. I think it should fail silently because ownership transfer can still work even without the email.
2. When the other person accepts the ownership transfer, their trial is ended immediately which causes that message to show up. This behaviour is only relevant for our Cloud product so it shouldn't do that in self-hosted. I'll fix it

**Comment by ukutaht** [comment]
I've released a patched version `v1.4.3` which should not generate these problems.

It won't fix the issue with the owner of your site. For that you need to access the database and remove their trial like so:

```
UPDATE users SET trial_expiry_date=null WHERE id=<new_owner_id>
```

**Comment by tmrwltd** [comment]
Ok. Thanks.
I also figured out why the mails not working: https://github.com/plausible/docs/issues/115

Everything works fine now.
